drivers/SmartThings/zigbee-button/src/test/test_wallhero_button.lua

```diff
@@ -0,0 +1,360 @@
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button25", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
+    test.socket.zigbee:__queue_receive({ mock_device.id, zigbee_test_utils.build_custom_command_id(mock_device, Scenes.ID, Scenes.server.commands.RecallScene.ID, 0x0000, "\x05\x00\x00\x00\x05\x00", 0x1A) })
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button26", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
+    test.socket.zigbee:__queue_receive({ mock_device.id, zigbee_test_utils.build_custom_command_id(mock_device, Scenes.ID, Scenes.server.commands.RecallScene.ID, 0x0000, "\x05\x00\x00\x00\x05\x00", 0x1B) })
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button27", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
+    test.socket.zigbee:__queue_receive({ mock_device.id, zigbee_test_utils.build_custom_command_id(mock_device, Scenes.ID, Scenes.server.commands.RecallScene.ID, 0x0000, "\x05\x00\x00\x00\x05\x00", 0x1C) })
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button28", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
+    test.socket.zigbee:__queue_receive({ mock_device.id, zigbee_test_utils.build_custom_command_id(mock_device, Scenes.ID, Scenes.server.commands.RecallScene.ID, 0x0000, "\x05\x00\x00\x00\x05\x00", 0x1D) })
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button29", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
+    test.socket.zigbee:__queue_receive({ mock_device.id, zigbee_test_utils.build_custom_command_id(mock_device, Scenes.ID, Scenes.server.commands.RecallScene.ID, 0x0000, "\x05\x00\x00\x00\x05\x00", 0x1E) })
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("button30", button_attr.pushed({ state_change = true }))
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", button_attr.pushed({ state_change = true }))
+    )
+    test.wait_for_events()
+
```

Comment:
ouch

for future reference, you could have used a loop here. e.g.: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-button/src/test/test_heiman_button.lua#L161